wsf_scraping/spiders/gov_spider.py

```diff
@@ -10,20 +11,47 @@ class GovSpider(BaseSpider):
         'JOBDIR': 'crawls/gov_uk'
     }
 
+    def __init__(self, *args, **kwargs):
+        year_before = kwargs.get('year_before', False)
+        year_after = kwargs.get('year_after', False)
+        id = kwargs.get('uuid', '')
+
+        self.uuid = id
+        if year_before:
+            self.year_before = year_before
+        else:
+            self.year_before = ''
+
+        if year_after:
+            self.year_after = year_after
+        else:
+            self.year_after = ''
+
     def start_requests(self):
         """Sets up the base urls and start the initial requests."""
-        urls = [
-            'https://www.gov.uk/government/publications',
-        ]
-
-        for url in urls:
-            self.logger.info('Initial url: %s', url)
-            yield scrapy.Request(
-                url=url,
-                callback=self.parse,
-                errback=self.on_error,
-                dont_filter=True,
-            )
+        url = 'https://www.gov.uk/government/publications'
+
+        if self.year_before or self.year_after:
+            query_dict = {
+                'keywords': '',
+                'publication_filter_option': 'all',
+                'topics[]': 'all',
+                'departments[]': 'all',
+                'official_document_status': 'all',
+                'world_locations[]': 'all',
+                'from_date': '01/01/{}'.format(self.year_after),
+                'to_date': '31/12/{}'.format(self.year_before)
+            }
+            query_params = urlencode(query_dict)
+            url = url + '?' + query_params
+
+        self.logger.info('Initial url: %s', url)
+        yield scrapy.Request(
+            url=url,
+            callback=self.parse,
+            errback=self.on_error,
+            dont_filter=True,
```

Comment:
`filter=False`?

Reply:
🤷‍♂️

Reply:
Scrapy internal 🤷‍♂️

Reply:
More seriously, Scrapy default behaviour is to filter everything, so it's a change to NOT filter things, and this is probably why the meta is `dont_filter`

Reply:
👍